File: leetcode/rust/find-the-pivot-integer.rs
impl Solution {
    pub fn pivot_integer(n: i32) -> i32 {
        let mut left = 0;
        let mut right = n * (n + 1) / 2;

        for i in 1..=n {
            left += i;

            if left == right {
                return i;
            }

            right -= i;
        }

        -1
    }
}

File: leetcode/python/find-the-pivot-integer.py
class Solution:
    def pivotInteger(self, n: int) -> int:
        left = 0
        right = n * (n + 1) / 2

        for i in range(1, n+1):
            left += i

            if left == right:
                return i
            
            right -= i
        
        return -1